### PR TITLE
Initial sync of refactor branch with master 

### PR DESCRIPTION
**Pull request by akashmjn**
* cleanup `download_datasets.py` : easy download of latest train/test sets 
* major fixes in baseline Pytorch/AudioSet training code (code hasn't been touched much since July '19) 
    * track train/validation set losses 
    * logging with tensorboard 
    * bugfix for optimizer/scheduler -> big perf improvement
    * clean progress bars 
    * support finetuning entire model 
    * averaging final model from multiple epochs 
* reorganize code & tweak README files a bit

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,89 @@
...
+```
+- data_ml 						(current directory)
+    train.py
+    test.py
+	- src						(module library)
+	- notebooks					(for evaluation, data preparation)
+    - tools
+- models
+- runs
+- live_inference				(deploy trained model)
```
Comment (prakruti) [inline README.md:24]: Remove these if they don't exist.

**Review by prakruti — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (akashmjn) [reply README.md:24]: Placeholder for when you migrate over code from Az devops
